### PR TITLE
Fix gpdb_icg job [#118185321]

### PR DESCRIPTION
**Pull request by oarap**
Refactor test_with_orca.py and build_with_orca.py
Extract the common piece out as GporcaCommon package.

Disable gpfdist tests with --disable-gpfdist option for configure,
because GPORCA doesn't impact the management utilities like gpfdist.

### CODE REVIEW

**Comment by oarap** [comment]
@d @hsyuan @vraghavan78 @edespino

File: concourse/scripts/build_with_orca.py [hunk 1]
```diff
@@ -98,14 +22,15 @@ def main():
     parser.add_option("--cxxflags", dest="cxxflags")
     parser.add_option("--output_dir", dest="output_dir", default="install")
     (options, args) = parser.parse_args()
-    status = install_system_deps()
+    myCommon = GporcaCommon()
```
Comment (hsyuan) [inline concourse/scripts/build_with_orca.py]: myCommon does not seem like a good var name.

File: concourse/scripts/gporca/GporcaCommon.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+             "/usr/local"])
+    
+    def configure(self):
+        return subprocess.call(["./configure",
+                                "--enable-orca",
+                                "--enable-mapreduce",
+                                "--with-perl",
+                                "--with-libxml",
+                                "--with-python",
+                                "--disable-gpfdist",
```
Comment (hsyuan) [inline concourse/scripts/gporca/GporcaCommon.py]: can we enable gpfdist?

File: concourse/scripts/test_with_orca.py [hunk 3]
```diff
@@ -125,17 +46,18 @@ def main():
     parser.add_option("--output_dir", dest="output_dir", default="install")
     parser.add_option("--gpdb_name", dest="gpdb_name")
     (options, args) = parser.parse_args()
-    status = install_system_deps()
+    myCommon = GporcaCommon()
```
Comment (hsyuan) [inline concourse/scripts/test_with_orca.py]: var naming

**Comment by hsyuan** [comment]
+1

**Pull request merged**